src/main/java/org/wildfly/security/keystore/LdapKeyStore.java

```diff
@@ -102,7 +103,7 @@ public LdapKeyStore build() {
                     filterAlias, filterCertificate, filterIterate, createPath, createRdn, createAttributes, aliasAttribute,
                     certificateAttribute, certificateType, certificateChainAttribute, certificateChainEncoding,
                     keyAttribute, keyType);
-            return new LdapKeyStore(spi, null, null);
+            return new LdapKeyStore(spi, new WildFlyElytronProvider(), "LdapRealm");
```

Comment:
I don't think this is the right thing to do.  We can't just instantiate another Elytron provider instance here.  Maybe we need some kind of NullProvider instance?